nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/DirConfigSource.java

```diff
@@ -269,22 +270,31 @@ public String getName() {
         return "Directory";
     }
 
-    Path findDir() throws IOException {
+    Optional<Path> findDir() throws IOException {
         String path = configService.getMPConfig().getSecretDir();
-        List<Path> candidates = new ArrayList<>();
+        if (path == null)
+            return Optional.empty();
         
         // adding all pathes where to look for the directory...
+        List<Path> candidates = new ArrayList<>();
         candidates.add(Paths.get(path));
-        // let's try it relative to server environment root
+        // let's try it relative to server environment root (<PAYARA-HOME>/glassfish/domains/<DOMAIN>/)
         if ( ! Paths.get(path).isAbsolute())
             candidates.add(Paths.get(System.getProperty("com.sun.aas.instanceRoot"), path).normalize());
         
         for (Path candidate : candidates) {
             if (isAptDir(candidate)) {
-                return candidate;
+                return Optional.of(candidate);
             }
         }
-        throw new IOException("Given MPCONFIG directory '"+path+"' is no directory, cannot be read or has a leading dot.");
+        
+        // Log that the configured directory is not resolving.
+        Level lvl = SEVERE;
+        // Reduce log level to info if default setting, as the admin might simply not use this functionality.
+        if(path.equals(DEFAULT_DIR))
+            lvl = INFO;
```

Comment:
I'd go even to `FINE` for this. Since this is default and directory is not present, there's no need for distraction in logs.

Reply:
My idea with using info: it would be easier to find out if it's not working as intended, in case someone actually wants to use it with the default value. Otherwise there won't be a message and people need to figure out what logging to tune.

I'll do as you folks handle this usually, please guide me 🤗

Reply:
There is already so much log entries during startup, we try not to introduce new ones unless they're necessary.

The default relative directory was also a bit of unfinished feature -- it wasn't documented and both docs and admin UI say it wants a full path. And I think it serves better when it is a full path as well. So I wouldn't advertise it too much by means of a log message at every deployment.

I think the best solution would be to remove default value, but that would break existing installations, therefore this one is the best way out we've got.

Reply:
:ok: 

Side note: the docs are updated in https://github.com/payara/Payara-Community-Documentation/pull/126 and now document the default value plus the option to go with relative path. One day there might be other location candidates where a relative path would be nice to use (that's why it's a list in `findDir()`)

Reply:
Done within 6921cab